CounterStrike/src/dev/danablend/counterstrike/CounterStrike.java

```diff
@@ -779,8 +831,18 @@ public CSPlayer getCSPlayer(Player player, boolean create, String colour) {
         }
 
         if (create) {
+            CSPlayer csp = new CSPlayer(this, player, colour);
+
+            //online filters NPCs
+            if (player.isOnline() && (ResourseHash.get(player.getName() + "RES") == null || ResourseHash.get(player.getName() + "RES") == "DEFAULT")) {

```

Comment:
It is better to use .equals() for String comparisons. see https://www.geeksforgeeks.org/difference-between-and-equals-method-in-java/